### PR TITLE
move to pulp3 to pulpcore feature

### PR DESCRIPTION
**Pull request by jlsherrill**

### CODE REVIEW

File: lib/smart_proxy_pulp_plugin/pulpcore_api.rb [hunk 1]
```diff
@@ -26,7 +26,7 @@ class Pulp3Api < Sinatra::Base
         size = :byte
       end
 
-      monitor_dirs = Hash[::PulpProxy::Pulp3Plugin.settings.marshal_dump.select { |key, _| key == :pulp_dir || key == :pulp_content_dir }]
```
Comment (ekohl) [inline lib/smart_proxy_pulp_plugin/pulpcore_api.rb]: I was sure we dropped disk usage but looks like we didn't merge https://github.com/theforeman/smart_proxy_pulp/pull/20. Perhaps together they make a nice 2.0?

Comment (jlsherrill) [reply lib/smart_proxy_pulp_plugin/pulpcore_api.rb]: sure, that sounds good

**Comment by jlsherrill** [comment]
FYI katello pr to adjust to this: https://github.com/Katello/katello/pull/8493

**Comment by jlsherrill** [comment]
actually since https://github.com/theforeman/smart_proxy_pulp/pull/20 has to be rebased, lets just merge this first

**Comment by jlsherrill** [comment]
eh, since you already rebased, we can do that first :)

**Comment by ekohl** [comment]
I figured that was easier and this should probably not be merged until the Katello PR at least has an ack.

File: lib/smart_proxy_pulp_plugin/pulpcore_plugin.rb [hunk 2]
```diff
@@ -15,9 +15,9 @@ class Pulp3Plugin < ::Proxy::Plugin
     expose_setting :mirror
     expose_setting :content_app_url
     capability( lambda do ||
```
Comment (ekohl) [inline lib/smart_proxy_pulp_plugin/pulpcore_plugin.rb]: I've been wondering if this should be simplified to:
```suggestion
     capability lambda { PulpcoreClient.capabilities }
```

File: test/pulpcore_api_test.rb [hunk 3]
```diff
@@ -3,19 +3,19 @@
...
+class PulpcoreApiTest < Test::Unit::TestCase
   include Rack::Test::Methods
 
   def app
-    PulpProxy::Pulp3Api.new
+    PulpProxy::PulpcoreApi.new
   end
 
   def test_returns_pulp_status_on_200
     PulpProxy::Pulp3Plugin.load_test_settings({})
```
Comment (ekohl) [inline test/pulpcore_api_test.rb]:
```suggestion
    PulpProxy::PulpcorePlugin.load_test_settings({})
```

**Comment by jlsherrill** [comment]
> Agreed this should be 2.0.0?

Agree  :100 %  !

File: lib/smart_proxy_pulp_plugin/pulpcore_plugin.rb [hunk 4]
```diff
@@ -14,7 +14,7 @@ class PulpcorePlugin < ::Proxy::Plugin
     expose_setting :pulp_url
     expose_setting :mirror
     expose_setting :content_app_url
-    capability( lambda do ||
+     capability lambda { PulpcoreClient.capabilities }
```
Comment (ekohl) [inline lib/smart_proxy_pulp_plugin/pulpcore_plugin.rb]: So the Github rendering was messing with me here

Comment (jlsherrill) [reply lib/smart_proxy_pulp_plugin/pulpcore_plugin.rb]: which old code did you mean?

Comment (ekohl) [reply lib/smart_proxy_pulp_plugin/pulpcore_plugin.rb]: Previously it was a multi line lambda. The GH suggestion only replaced the first line, keeping the second and third line. That broke the build. It's fine (and probably better) to keep this PR minimal and clean up the lambda later.

**Comment by jlsherrill** [comment]
@ekohl if you're good with this, i propose that we merge this when https://github.com/Katello/katello/pull/8493 goes Green.  

I can then do a 2.0 release of this plugin.

**Comment by ekohl** [comment]
Sounds like a great plan.

**Pull request merged**